pf/tests/integration/program_test.go

```diff
@@ -107,6 +107,31 @@ func TestDefaultInfo(t *testing.T) {
 	})
 }
 
+func TestPrivateState(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("Skipping on Windows due to a PATH setup issue where the test cannot find pulumi-resource-testbridge.exe")
+	}
+
+	wd, err := os.Getwd()
+	assert.NoError(t, err)
+	bin := filepath.Join(wd, "..", "bin")
+
+	integration.ProgramTest(t, &integration.ProgramTestOptions{
+		Env:                    []string{fmt.Sprintf("PATH=%s", bin)},
+		Dir:                    filepath.Join("..", "testdata", "privst-program"),
+		ExtraRuntimeValidation: validateExpectedVsActual,
+		SkipRefresh:            true,
+		RequireService:         true, // EditDirs do not work otherwise
```

Comment:
Is there a pu/pu bug associated with this? I'm not aware of a structural reason why this shouldn't work.

Reply:
Probably https://github.com/pulumi/pulumi/issues/12859